initialise/MirahezeFunctions.php

```diff
@@ -0,0 +1,375 @@
+						'u' => $explode[2] ?? $explode[0] . '.' . ( $_SERVER['HTTP_HOST'] ?? 'miraheze.org' ),
+						's' => $explode[3] ?? 'No sitename set.',
+					];
+				}
+			}
+		}
+
+		if ( $onlyDBs ) {
+			return array_keys( $databases );
+		}
+
+		return $databases;
+	}
+
+	public static function getRealm() {
+		$domain = explode( '.', $_SERVER['SERVER_NAME'], 2 )[1];
+
+		return self::REALMS[$domain] ?? 'default';
+	}
+
+	public function getCurrentDatabase() {
+		if ( defined( 'MW_DB' ) ) {
+			return MW_DB;
+		}
+
+		$explode = explode( '.', $this->hostname, 2 );
+
+		if ( $explode[0] == 'www' ) {
+			$explode = explode( '.', $explode[1], 2 );
+		}
+
+		foreach ( self::SUFFIXES as $suffix => $site ) {
+			if ( $explode[1] == $site ) {
+				return $explode[0] . $suffix;
+			}
+		}
+	}
+
+	public function setDatabase() {
+		global $wgConf;
+
+		$wgConf->settings['wgDBname'][$this->dbname] = $this->dbname;
+	}
+
+	public function getDatabaseClusters() {
+		$allDatabases = self::readDbListFile( 'all', false );
+		$deletedDatabases = self::readDbListFile( 'deleted', false );
+
+		$databases = array_merge( $allDatabases, $deletedDatabases );
+
+		$clusters = array_column( $databases, 'c' );
+
+		return array_combine( array_keys( $databases ), $clusters );
+	}
+
+	public function getServer() {
+		global $wgConf;
+
+		$wgConf->settings['wgServer']['default'] = 'https://' . self::SUFFIXES[ array_key_first( self::SUFFIXES ) ];
+
+		$databases = self::readDbListFile( 'all', false );
+		foreach ( $databases as $db => $data ) {
+			foreach ( self::SUFFIXES as $suffix ) {
+				if ( substr( $db, -strlen( $suffix ) ) == $suffix ) {
+					$wgConf->settings['wgServer'][$db] = $data['u'] ?? 'https://' . substr( $db, 0, -strlen( $suffix ) ) . '.' . self::SUFFIXES[$suffix];
+				}
+			}
+		}
+
+		return $wgConf->settings['wgServer'][$this->dbname] ??
+			$wgConf->settings['wgServer']['default'];
+	}
+
+	public function getSitename() {
+		global $wgConf;
+
+		$allDatabases = self::readDbListFile( 'all', false );
+		$deletedDatabases = self::readDbListFile( 'deleted', false );
+
+		$databases = array_merge( $allDatabases, $deletedDatabases );
+
+		$siteNameColumn = array_column( $databases, 's' );
+
+		$siteNames = array_combine( array_keys( $databases ), $siteNameColumn );
+		$siteNames['default'] = 'No sitename set.';
+
+		$wgConf->settings['wgSitename'] = $siteNames;
+
+		return $siteNames[$this->dbname];
+	}
+
+	public function isMissing() {
+		global $wgConf;
+
+		return !in_array( $this->getCurrentDatabase(), $wgConf->wikis );
+	}
+
+	public function getCacheArray() {
+		global $wgCreateWikiCacheDirectory;
+
+		// If we don't have a cache file, let us exit here
+		if ( !file_exists( $wgCreateWikiCacheDirectory . '/' . $this->dbname . '.json' ) ) {
+			return false;
+		}
+
+		$this->cacheArray = (array)json_decode( file_get_contents( $wgCreateWikiCacheDirectory . '/' . $this->dbname . '.json' ), true );
+
+		return (array)$this->cacheArray;
```

Comment:
```suggestion
		return $this->cacheArray;
```

Possibly redundant casting?

Reply:
Yep, sure is. Thanks!